x-pack/plugins/session_view/public/components/tty_player/hooks.ts

```diff
@@ -97,16 +92,13 @@ export const useIOLines = (pages: ProcessEventsPage[] | undefined) => {
               const splitLines = process.io.text.split(TTY_LINE_SPLITTER_REGEX);
               const combinedLines = [splitLines[0]];
               const previousProcessId = previous[previous.length - 1]?.event.process?.entity_id;
-              const currentProcessLineInfo: ProcessEntityIdIOLine = {
-                value: entityIdLineMap[process.entity_id]?.value ?? previous.length,
-              };
 
-              if (previousProcessId && previousProcessId !== process.entity_id) {
-                entityIdLineMap[previousProcessId].next = currentProcessLineInfo.value;
-                currentProcessLineInfo.previous = entityIdLineMap[previousProcessId].value;
-              }
               if (previousProcessId !== process.entity_id) {
-                entityIdLineMap[process.entity_id] = currentProcessLineInfo;
+                const processLineInfo: ProcessStartMarker = {
+                  line: previous.length || 0,
+                  event,
+                };
+                processStartMarkers.push(processLineInfo);
```

Comment:
@opauloh I changed this to an array. The use of map didn't make sense as it would treat each process as a single instance. A marker should be created anytime the current process changes. This should make your end a bit easier hopefully.

Reply:
That makes sense, and it will help!